### PR TITLE
fix dup registry on OSP

### LINKED ISSUE #164: openstack provisioning, using cinder for registry volume, should work with multiple clusters
When provisioning multiple clusters within the same tenant, the volume names used to back the registry should be unique (i.e.: `env_id` and `dns_domain`)

### PR DESCRIPTION
**Pull request by day4skiing**
#### What does this PR do?
This PR automatically generates a Unique name per cluster when automatic provisioning on OSP is requested

#### How should this be manually tested?
Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples)
Define the following in `all.yml` inventory and run `end-to-end.yml`
```
# enable automatic provisioning of cinder registry volume on OSP
casl_storage_registry_kind: 'cinder'
casl_storage_registry_create: True
```
A registry volume defaulted to 20G will be created with the following name `{{ env_id }}.{{ dns_domain }}-registry` which must be unique per tenant.

#### Is there a relevant Issue open for this?
resolves #164 

#### Who would you like to review this?
cc: @redhat-cop/casl @oybed

### CODE REVIEW

File: docs/PROVISIONING_OPENSTACK.md [hunk 1]
```diff
@@ -53,18 +53,10 @@ The following is just an example on how the `sample.osp.example.com` inventory c
...
-osp_volumes:
-- name: "registry-vol"
-  display_description: "Registry Volume"
-  display_name: "registry-vol"
-  size: 50
-  purpose: "ose_registry"
-  ```
-> **Note 1:** `openshift_hosted_registry_storage_volume_size` and `openshift_hosted_registry_storage_openstack_volumeID` are automatically set during play execution
+casl_storage_registry_kind: 'cinder'
+casl_storage_registry_create: True
```
Comment (day4skiing) [inline docs/PROVISIONING_OPENSTACK.md]: Not sure this flag is necessary

Comment (oybed) [reply docs/PROVISIONING_OPENSTACK.md]: @day4skiing you really don't need any of these ... what you had for this was just fine

File: inventory/sample.osp.example.com.d/inventory/group_vars/OSEv3.yml [hunk 2]
```diff
@@ -56,6 +56,15 @@ openshift_metrics_cassandra_storage_type: dynamic
 openshift_metrics_cassandra_pvc_size: 40G
 openshift_metrics_duration: 2
 
+# Cinder registry storage
```
Comment (oybed) [inline inventory/sample.osp.example.com.d/inventory/group_vars/OSEv3.yml]: keep these in `all.yml` as otherwise the post-provision-openstack tools won't find them

File: playbooks/openshift/post-provision-openstack.yml [hunk 3]
```diff
@@ -18,7 +18,6 @@
 # provision cinder volume
 - import_playbook: cinder-registry.yml
   when:
-  - openshift_hosted_registry_storage_kind is defined
```
Comment (oybed) [inline playbooks/openshift/post-provision-openstack.yml]: what you had here worked just fine - not sure why the change? I'd say keep it as-is and just key off the `openshift_hosted_registry_storage_kind` and `openshift_hosted_registry_storage_openstack_volumeID` - i.e.: if the latter is NOT set, we'll provision one to be used ... again, that's the way you had it and should work fine as-is

File: inventory/sample.osp.example.com.d/inventory/group_vars/all.yml [hunk 4]
```diff
@@ -93,18 +93,7 @@ rhsm_password: ''
...
-osp_volumes:
-- name: "registry-vol"
-  display_description: "Registry Volume"
-  display_name: "registry-vol"
-  size: 50
-  purpose: "ose_registry"
+# enable automatic provisioning of cinder registry volume on OSP
+casl_storage_registry_kind: 'cinder'
+casl_storage_registry_create: True
+#casl_registry_size: 20
```
Comment (oybed) [inline inventory/sample.osp.example.com.d/inventory/group_vars/all.yml]: you can use the `openshift_hosted_registry_storage_volume_size` inventory variable instead of making a `casl` specific one

File: playbooks/openshift/cinder-registry.yml [hunk 5]
```diff
@@ -1,20 +1,28 @@
 --- 
 # Create a volume for the registry
 - hosts: localhost
-  roles: 
+  vars:
+    osp_volumes:
+    - name: "{{ env_id }}.{{ dns_domain }}-registry"
```
Comment (oybed) [inline playbooks/openshift/cinder-registry.yml]: consider using `full_dns_domain` here (and below) - as seen [here](https://github.com/redhat-cop/casl-ansible/blob/master/playbooks/openshift/pre-tasks.yml#L25)

File: playbooks/openshift/cinder-registry.yml [hunk 6]
```diff
@@ -1,20 +1,28 @@
...
+  roles:
   - infra-ansible/roles/osp/admin-volume
 
 - hosts: OSEv3
   tasks:
-  - name: "set registry volumeID fact"
-    set_fact: 
+  - name: "set registry fact"
+    set_fact:
+      openshift_hosted_registry_storage_kind: openstack
```
Comment (oybed) [inline playbooks/openshift/cinder-registry.yml]: this should already be set in the inventory, so no need to set it here

File: playbooks/openshift/cinder-registry.yml [hunk 7]
```diff
@@ -1,20 +1,28 @@
...
       openshift_hosted_registry_storage_openstack_volumeID: "{{ disk.volume.id }}"
       openshift_hosted_registry_storage_volume_size: "{{ disk.volume.size }}Gi"
-    with_items: 
+      openshift_hosted_registry_storage_access_modes: ['ReadWriteOnce']
```
Comment (oybed) [inline playbooks/openshift/cinder-registry.yml]: Let's only set these two if they are not already set...

File: inventory/sample.osp.example.com.d/inventory/group_vars/OSEv3.yml (hunk 2 above)
Comment (day4skiing) [reply inventory/sample.osp.example.com.d/inventory/group_vars/OSEv3.yml]: If we keep these in `all.yml` I believe it conflicts with the implementation in [openshift-ansible-contrib](https://github.com/openshift/openshift-ansible-contrib/blob/master/playbooks/provisioning/openstack/advanced-configuration.md#use-an-existing-cinder-volume-for-the-openshift-registry)  causing the issue described in  #162

Comment (oybed) [reply inventory/sample.osp.example.com.d/inventory/group_vars/OSEv3.yml]: @day4skiing no, there's no conflict with `contrib` for this. Basically the `post` step doesn't have the necessary values to proceed and hence the individual playbooks cannot be used. Anyway - let's address that in a separate PR to avoid too much cross referencing.

**Comment by day4skiing** [comment]
PTAL

**Comment by oybed** [comment]
@day4skiing I added the `resolves` keyword to the description to auto-close out the issue

File: docs/PROVISIONING_OPENSTACK.md [hunk 8]
```diff
@@ -46,17 +46,7 @@ The following is just an example on how the `sample.osp.example.com` inventory c
 2) Edit `~/src/casl-ansible/inventory/sample.osp.example.com.d/inventory/group_vars/all.yml` to match your cloud provider/environment/project/tenant. See comments in the file for more detailed information on how to fill these in. To utilize OSP cinder storage for the registry add the following to `all.yml`
 ```
 openshift_hosted_registry_storage_kind: openstack
-openshift_hosted_registry_storage_access_modes: ['ReadWriteOnce']
```
Comment (oybed) [inline docs/PROVISIONING_OPENSTACK.md]: Please include `openshift_hosted_registry_storage_volume_size` here and explain that it can be used to set a proper size. (maybe with an indication that `20` is the default)

File: inventory/sample.osp.example.com.d/inventory/group_vars/all.yml [hunk 9]
```diff
@@ -93,18 +93,14 @@ rhsm_password: ''
...
-
-osp_volumes:
-- name: "registry-vol"
-  display_description: "Registry Volume"
-  display_name: "registry-vol"
-  size: 50
-  purpose: "ose_registry"
+#openshift_hosted_registry_storage_volume_size: 20
+
+# User provisioned Cinder registry storage
```
Comment (oybed) [inline inventory/sample.osp.example.com.d/inventory/group_vars/all.yml]: Please elaborate to include info indicating that the below will leverage an *existing* Cinder volume with the provided VolumeID.

**Comment by oybed** [comment]
@day4skiing much better - just a few documentation updates and we can merge.
